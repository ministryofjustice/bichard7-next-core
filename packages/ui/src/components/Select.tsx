import type { ComponentProps } from "react"
import { mergeClassNames } from "helpers/mergeClassNames"

export interface SelectProps extends ComponentProps<"select"> {
  showError?: boolean
  placeholder?: string
}

export const Select = ({ placeholder, showError, className, children, ...props }: SelectProps) => {
  return (
    <select {...props} className={mergeClassNames("govuk-select", showError ? "govuk-select--error" : "", className)}>
      {placeholder ? (
        <option disabled hidden value="">
          {placeholder}
        </option>
      ) : null}
      {children}
    </select>
  )
}
