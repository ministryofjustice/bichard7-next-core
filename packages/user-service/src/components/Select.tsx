export type Option = {
  name: string
  id: number | string
}

interface Props {
  options?: Option[]
  label: string
  id: string
  defaultValue?: number | string
}

const Select = ({ options, label, id, defaultValue }: Props) => (
  <div className="govuk-form-group">
    <label className="govuk-label govuk-!-padding-top-2" htmlFor={id}>
      {label}
    </label>
    <select className="govuk-select" id={id} name={id}>
      {options &&
        options.map((option: Option) => (
          <option selected={String(option.id) === String(defaultValue)} key={option.name} value={option.id}>
            {option.name}
          </option>
        ))}
    </select>
  </div>
)

export default Select
