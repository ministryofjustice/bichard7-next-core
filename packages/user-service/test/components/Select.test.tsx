import { render } from "@testing-library/react"
import Select, { Option } from "components/Select"

it("should render select component correctly", () => {
  const options: Option[] = [
    {
      id: "test-id-01",
      name: "test-name-01"
    },
    {
      id: "test-id-02",
      name: "test-name-02"
    },
    {
      id: "test-id-03",
      name: "test-name-03"
    }
  ]

  const { container } = render(<Select options={options} label="test-select-label" id="test-id" />)

  expect(container).toMatchSnapshot()
})
