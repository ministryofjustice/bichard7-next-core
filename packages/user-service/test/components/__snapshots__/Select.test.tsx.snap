// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`should render select component correctly 1`] = `
<div>
  <div
    class="govuk-form-group"
  >
    <label
      class="govuk-label govuk-!-padding-top-2"
      for="test-id"
    >
      test-select-label
    </label>
    <select
      class="govuk-select"
      id="test-id"
      name="test-id"
    >
      <option
        value="test-id-01"
      >
        test-name-01
      </option>
      <option
        value="test-id-02"
      >
        test-name-02
      </option>
      <option
        value="test-id-03"
      >
        test-name-03
      </option>
    </select>
  </div>
</div>
`;
